src/runway/blueprints/staticsite/staticsite.py

```diff
@@ -100,57 +103,248 @@ class StaticSite(Blueprint):  # pylint: disable=too-few-public-methods
+            'DefaultRootObject': 'index.html',
+            'Logging': If(
+                'CFLoggingEnabled',
+                cloudfront.Logging(
+                    Bucket=Join('.',
+                                [variables['LogBucketName'].ref,
+                                 's3.amazonaws.com'])
+                ),
+                NoValue
+            ),
+            'PriceClass': variables['PriceClass'].ref,
+            'Enabled': True,
+            'WebACLId': If(
+                'WAFNameSpecified',
+                variables['WAFWebACL'].ref,
+                NoValue
+            ),
+            'ViewerCertificate': If(
+                'AcmCertSpecified',
+                cloudfront.ViewerCertificate(
+                    AcmCertificateArn=variables['AcmCertificateArn'].ref,
+                    SslSupportMethod='sni-only'
+                ),
+                NoValue
+            )
+        }
+
+    def add_origin_access_identity(self):
+        """Add the origin access identity resource to the template.
+
+        Returns:
+            dict: The OAI resource
+
+        """
+        return self.template.add_resource(
             cloudfront.CloudFrontOriginAccessIdentity(
                 'OAI',
+                Condition='CFEnabled',
                 CloudFrontOriginAccessIdentityConfig=cloudfront.CloudFrontOriginAccessIdentityConfig(  # noqa pylint: disable=line-too-long
                     Comment='CF access to website'
                 )
             )
         )
 
-        bucket = template.add_resource(
+    def add_bucket_policy(self, bucket):
+        """Add a policy to the bucket if CloudFront is disabled. Ensure PublicRead.
+
+        Keyword Args:
+            bucket (dict): The bucket resource to place the policy
+
+        Returns:
+            dict: The Bucket Policy Resource
+
+        """
+        return self.template.add_resource(
+            s3.BucketPolicy(
+                'BucketPolicy',
+                Bucket=bucket.ref(),
+                Condition='CFDisabled',
+                PolicyDocument=Policy(
+                    Version="2012-10-17",
+                    Statement=[
+                        Statement(
+                            Effect=Allow,
+                            Principal=Principal('*'),
+                            Action=[Action('s3', 'getObject')],
+                            Resource=[
+                                Join('', ['arn:aws:s3:::',
+                                          bucket.ref(),
+                                          '/*'])
+                            ],
+                        )
+                    ]
+                )
+            )
+        )
+
+    def add_bucket(self):
+        """Add the bucket resource along with an output of it's name.
+
+        Returns:
+            dict: The bucket resource
+
+        """
+        variables = self.get_variables()
+        access = s3.PublicRead if (variables['CFDisabled'] == 'true') else s3.Private
```

Comment:
This should be a CloudFormation If (so it's evaluated at stack deployment time) on AccessControl, not a python if.

Will look something like `AccessControl=If('CFEnabled', s3.Private, s3.PublicRead)`